test/suite_oc-desktop/shared/steps/steps.py

```diff
@@ -0,0 +1,343 @@
+    mouseClick(waitForObject(names.leUrl_OCC_PostfixLineEdit), 0, 0, Qt.NoModifier, Qt.LeftButton)
+    type(waitForObject(names.leUsername_QLineEdit), user)
+    type(waitForObject(names.leUsername_QLineEdit), "<Tab>")
+    type(waitForObject(names.lePassword_QLineEdit), password)
+    clickButton(waitForObject(names.owncloudWizard_qt_passive_wizardbutton1_QPushButton))
+    clickButton(waitForObject(names.pbSelectLocalFolder_QPushButton))
+    mouseClick(waitForObject(names.fileNameEdit_QLineEdit), 0, 0, Qt.NoModifier, Qt.LeftButton)
+    type(waitForObject(names.fileNameEdit_QLineEdit), localfolder)
+    clickButton(waitForObject(names.qFileDialog_Choose_QPushButton))
+    clickButton(waitForObject(names.owncloudWizard_qt_passive_wizardbutton1_QPushButton))
+
+def isItemSynced(type, itemName):
+    if type != 'FILE' and type != 'FOLDER':
+        raise Exception("type must be 'FILE' or 'FOLDER'")
+    socketConnect = syncstate.SocketConnect()
+    socketConnect.sendCommand("RETRIEVE_" + type + "_STATUS:" + itemName + "\n");
+
+    if not socketConnect.read_socket_data_with_timeout(0.1):
+        return False
+    for line in socketConnect.get_available_responses():
+        if (line.startswith('STATUS:OK') and line.endswith(itemName)):
+            return True
+        elif line.endswith(itemName):
+            return False
+
+def isFolderSynced(folderName):
+    return isItemSynced('FOLDER', folderName)
+
+def isFileSynced(fileName):
+    return isItemSynced('FILE', fileName)
+
+def waitForFileToBeSynced(context, fileName):
+    waitFor(
+        lambda: isFileSynced(context.userData['clientSyncPath'] + fileName),
+        context.userData['clientSyncTimeout'] * 1000
+    )
+
+def substituteInLineCodes(context, value):
+    from urllib.parse import urlparse
+    value = value.replace('%local_server%', context.userData['localBackendUrl'])
+    value = value.replace('%client_sync_path%', context.userData['clientSyncPath'])
+    value = value.replace('%local_server_hostname%', urlparse(context.userData['localBackendUrl']).netloc)
+
+    return value
+
+def shareResource(resource):
+    socketConnect = syncstate.SocketConnect()
+    socketConnect.sendCommand("SHARE:" + resource + "\n")
+    if not socketConnect.read_socket_data_with_timeout(0.1):
+        return False
+    for line in socketConnect.get_available_responses():
+        if (line.startswith('SHARE:OK') and line.endswith(resource)):
+            return True
+        elif line.endswith(resource):
+            return False
+
+def executeStepThroughMiddleware(context, step, errorMessage=''):
+    body = {
+    "step": step}
+    params = json.dumps(body).encode('utf8')
+        
+    req = urllib.request.Request(
+        context.userData['middlewareUrl'] + 'execute',
+        data=params,
+        headers={"Content-Type": "application/json"}, method='POST'
+    )
+    res = urllib.request.urlopen(req)
+    if res.getcode() != 200:
```

Comment:
if something fails on the middleware it returns the error stack on the body, you can log that here for easier debug

Reply:
done in ae37abdb130b835becbbbba6215190f16030803e